linkerd/app/core/src/control.rs

```diff
@@ -89,7 +84,12 @@ impl Config {
             .push(tls::Client::layer(identity))
             .push_timeout(self.connect.timeout)
             .push(self::client::layer())
-            .push(reconnect::layer(connect_backoff))
+            .push_on_response(svc::MapErrLayer::new(Into::into))
+            .into_new_service()
+            .push(svc::NewReconnect::layer({
+                let backoff = self.connect.backoff;
+                move |_| Ok(backoff.stream())
+            }))
```

Comment:
nit/TIOLI: it looks like every `NewReconnect` is called with a closure that takes the `backoff` and calls `stream()` on it...should the `layer` fn just take something implementing `Backoff`?

not a blocker either way.